### PR TITLE
Add createGridFSBucket API.

### PR DESCRIPTION
**Pull request by davidlehn**
Alternate approach to https://github.com/digitalbazaar/bedrock-mongodb/pull/17.
Add an API to create a GridFSBucket.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -349,6 +349,25 @@ api.createIndexes = (options, callback) => async.each(
...
+ *
+ * By default the writeOptions config value is used for the GridFSBucket
+ * writeConcern option.
+ *
+ * @param options see GridFSBucket documentation
+ * @param callback(err, bucket) called once the operation completes.
+ */
+api.createGridFSBucket = (options, callback) => {
+  let opts = options;
+  if(!('writeConcern' in options)) {
```
Comment (mattcollier) [inline lib/index.js]: could this use `_.defaults` and avoid all the opts/options gyrations?  Although `lodash` is not required in this file, it's already a dependency.

Comment (davidlehn) [reply lib/index.js]: I was micro optimizing for no big reason.  Will simplify.

File: lib/index.js [hunk 2]
```diff
@@ -349,6 +349,20 @@ api.createIndexes = (options, callback) => async.each(
...
+ * Creates a streaming GridFS bucket interface.
+ *
+ * By default the writeOptions config value is used for the GridFSBucket
+ * writeConcern option.
+ *
+ * @param options see GridFSBucket documentation
+ * @param callback(err, bucket) called once the operation completes.
+ */
+api.createGridFSBucket = (options, callback) => {
+  const opts = Object.assign({}, {writeConerrn: api.writeOptions}, options);
```
Comment (mattcollier) [inline lib/index.js]: typo

Comment (davidlehn) [reply lib/index.js]: Oops, fixed.

File: lib/index.js [hunk 3]
```diff
@@ -349,6 +349,20 @@ api.createIndexes = (options, callback) => async.each(
...
+/**
+ * Creates a streaming GridFS bucket interface.
+ *
+ * By default the writeOptions config value is used for the GridFSBucket
+ * writeConcern option.
+ *
+ * @param options see GridFSBucket documentation
+ * @param callback(err, bucket) called once the operation completes.
+ */
+api.createGridFSBucket = (options, callback) => {
```
Comment (mattcollier) [inline lib/index.js]: Why is this async?

Comment (davidlehn) [reply lib/index.js]: Hmm.  I did have a weak reason for this, but it doesn't need to be.  I'll change it.

**Pull request merged**